python/tvm/topi/cuda/tensorcore_alter_op.py

```diff
@@ -170,7 +172,19 @@ def _dense_legalize(attrs, inputs, arg_types):
 
     if extra_flops_ratio > 2:
         logger.info("dense pad_to_tensorcore skipped, extra_flops_ratio %s", extra_flops_ratio)
-        return None
+
+        # If tensorcore schedule padding fails, pad to nearest upward 4x4x4 as long as
+        # the additional flops ratio isn't double or more.
+        # Note that 4x4x4 is invalid for tensorcore scheduling, but padding upwards to 4x4x4
+        # doesn't hurt if tensorcore padding has already failed.
+        if M % 4 == 0 and K % 4 == 0 and N % 4 == 0:
+            # No need to pad
+            return None
+        (dm, dk, dn) = _pad_to(M, K, N, (4, 4, 4))
+        extra_flops_ratio = _extra_flops(M, K, N, dm, dk, dn) / (M * K * N)
+
+        if extra_flops_ratio > 2:
+            return None
```

Comment:
@masahi I moved the 4x4x4 treatment to this block. Is this what you meant or did you want the 4x4x4 logic outside of `tensorcore_alter_op.py` entirely?

Reply:
Yes I think this is good. I don't like this file being named `tensorcore_alter_op.py` but that's a different story.

But I think this code block is only useful for int8. So probably we need `if dtype in ["int8", "uint8"]` check? Also I'm not clear what happens if padding is necessary but `extra_flops_ratio > 2` says True.

Reply:
And please do not nest `if extra_flops_ratio > 2:` check, since it's a bit weird.

Reply:
Sounds good.

Reply:
So I suggest something like this:

```
do_pad_for_int8 = ...

if extra_flops_ratio > 2 and not do_pad_for_int8:
     logger.info(...)
     return None
```

Reply:
Refactored the return None to be ordered after all the logic for dense slightly differently, PTAL